play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -997,7 +997,7 @@ class BattleForfeitPanel extends PSRoomPanel {
 				</p>
 				<p>
 					<button type="submit" class="button"><strong>Forfeit</strong></button> {}
-					{!battleRoom.battle.rated && <button type="button" data-href="replaceplayer" class="button">
+					{!battleRoom?.battle?.rated && <button type="button" data-href="replaceplayer" class="button">
```

Comment:
Why `?`?

I think crashing when something that shouldn't happen happens is fine. It'll just make it easier to track down the bug later.